src/core/server/saved_objects/migrations/types.ts

```diff
@@ -39,10 +39,10 @@ import { SavedObjectsMigrationLogger } from './core/migration_logger';
  *
  * @public
  */
-export type SavedObjectMigrationFn = (
-  doc: SavedObjectUnsanitizedDoc,
+export type SavedObjectMigrationFn<InputProps = any, MigratedProps = any> = (
```

Comment:
This would probably require updating the types of many migrations, but is it possible to default to `unknown` instead of `any`? I think that would encourage more plugins to type their migrations and make it explicit that their migration is dangerously untyped.

What do you think about calling these generic arguments `InputAttributes` and `MigratedAttributes` since they define the `attributes` property?

Reply:
> This would probably require updating the types of many migrations, but is it possible to default to unknown instead of any

I wanted to avoid impacting plugin code with that change TBH, therefor the defaulting `any`. But it may be better to still do it now to encourage yet-to-migrate SO migrations to be explicitly typed. 

> What do you think about calling these generic arguments InputAttributes and MigratedAttributes since they define the attributes property?

Indeed better. I always mix up `attributes` and `props` terminology for SOs.